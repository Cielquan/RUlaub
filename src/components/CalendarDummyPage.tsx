import { Trans } from "@lingui/macro";
import { StaticDatePicker } from "@mui/lab";
import { TextField, Typography } from "@mui/material";
import { Box } from "@mui/system";
import React, { ReactElement } from "react";
import { useDispatch } from "react-redux";
import { bindActionCreators } from "redux";

import { actionCreators } from "../state";

const CalendarDummyPage = (): ReactElement => {
  const dispatch = useDispatch();
  const { updateLocalConfig } = bindActionCreators(actionCreators, dispatch);

  const [date, setDate] = React.useState<Date>();

  const handleChange = (newDate: Date | null): void => {
    if (newDate !== null) {
      updateLocalConfig({ settings: { yearToShow: newDate.getFullYear() } });
      setDate(newDate);
    }
  };

  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        textAlign: "center",
      }}
    >
      <Typography variant="h4" sx={{ paddingTop: 4, paddingBottom: 8 }}>
        <Trans>No year to show selected. Please select one below:</Trans>
      </Typography>
      <StaticDatePicker
        displayStaticWrapperAs="desktop"
        views={["year"]}
        value={date}
        allowSameDateSelection
        autoFocus
        onChange={handleChange}
        // eslint-disable-next-line react/jsx-props-no-spreading
        renderInput={(params) => <TextField {...params} />}
      />
    </Box>
  );
};

export default CalendarDummyPage;
