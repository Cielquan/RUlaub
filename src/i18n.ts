import { i18n } from "@lingui/core";
import { de, en } from "make-plural/plurals";
import { messages as deMessages } from "./locales/de-DE/messages";
import { messages as enMessages } from "./locales/en-US/messages";

export type SupportedLocales = "de-DE" | "en-US";
export type SupportedLocalesShorted = "deDE" | "enUS";

export interface Language {
  name: string;
  locale: SupportedLocales;
  importName: SupportedLocalesShorted;
}

export const Languages: { [key: string]: Language } = {
  german: {
    name: "Deutsch",
    locale: "de-DE",
    importName: "deDE",
  },
  english: {
    name: "English",
    locale: "en-US",
    importName: "enUS",
  },
};

export const defaultLocale: SupportedLocales = "en-US";
export const defaultLanguage: Language = Object.values(Languages).filter(
  (lang) => lang.locale === defaultLocale
)[0];

i18n.loadLocaleData({
  "de-DE": { plurals: de },
  "en-US": { plurals: en },
});
i18n.load("de-DE", deMessages);
i18n.load("en-US", enMessages);
i18n.activate(defaultLocale);

export default Languages;
