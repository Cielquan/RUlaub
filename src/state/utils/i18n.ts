export type SupportedLocales = "de-DE" | "en-US";
export type SupportedLocalesShorted = "deDE" | "enUS";

export interface Language {
  name: string;
  locale: SupportedLocales;
  importName: SupportedLocalesShorted;
}

export const Languages: { [key: string]: Language } = {
  german: {
    name: "Deutsch",
    locale: "de-DE",
    importName: "deDE",
  },
  english: {
    name: "English",
    locale: "en-US",
    importName: "enUS",
  },
};

export default Languages;
