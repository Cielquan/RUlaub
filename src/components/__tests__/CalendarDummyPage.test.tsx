import { I18nProvider } from "@lingui/react";
import React from "react";
import renderer from "react-test-renderer";

import i18n from "../../i18n";

import CalendarDummyPage from "../CalendarDummyPage";

describe("<CalendarDummyPage />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <I18nProvider i18n={i18n}>
          <CalendarDummyPage />
        </I18nProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
