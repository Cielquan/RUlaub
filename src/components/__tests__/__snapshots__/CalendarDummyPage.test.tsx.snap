// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarStartPage /> renders correctly 1`] = `
<div
  className="MuiBox-root css-xi606m"
>
  <h4
    className="MuiTypography-root MuiTypography-h4 css-vhedur-MuiTypography-root"
  >
    No year to show selected.
  </h4>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Choose date, selected date is Sep 10, 2021"
        aria-readonly={true}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        readOnly={true}
        required={false}
        type="text"
        value="2021"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1a1ihhj"
        >
          <span
            className="notranslate"
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
